iceoryx2-bb/container/src/byte_string.rs

```diff
@@ -242,22 +242,13 @@ impl<const CAPACITY: usize> FixedSizeByteString<CAPACITY> {
     pub fn from_bytes_truncated(bytes: &[u8]) -> Self {
         let mut new_self = Self::new();
         new_self.len = std::cmp::min(bytes.len(), CAPACITY);
-        unsafe {
-            std::ptr::copy(
-                bytes.as_ptr(),
-                new_self.data.as_ptr() as *mut u8,
-                bytes.len(),
-            )
-        };
+        for i in 0..new_self.len {
+            new_self.data[i].write(bytes[i]);
+        }
 
-        let zero = 0u8;
-        unsafe {
-            std::ptr::copy(
-                &zero,
-                new_self.data.as_ptr().add(new_self.len) as *mut u8,
-                1,
-            )
-        };
+        if new_self.len < CAPACITY {
+            new_self.data[new_self.len].write(0);
+        }
```

Comment:
Out of curiosity. If the 0 is not strictly required, why add it at all?

If it is required then it should be added unconditional.

Reply:
Since there is a method `as_bytes_with_nul`. I could of course add the `0` only there whenever it is called, but for me, it seemed to be cleaner to add the `0` right from the beginning.
And I need at least reserve the space for an additional `0` since `as_bytes_with_nul` must return a consecutive memory range with a terminating zero.

Reply:
Oh, and this method is used in a lot of posix wrappers.

Reply:
Writing a 0 only when required is a bad idea. See also https://www.youtube.com/watch?v=kPR8h4-qZdk&t=1150s ... although the implementation in Rust would probably not lead to the issue described in the talk.